docs/designers-developers/developers/data/data-core-edit-post.md

```diff
@@ -339,6 +339,24 @@ Returns an action object used to toggle a plugin name flag.
 
  * pluginName: Plugin name.
 
+### disableBlockTypes
+
+Returns an action object used in signalling that block types by the given
+name(s) should be disabled.
+
+*Parameters*
+
+ * blockNames: Names of block types to disable.
+
+### enableBlockTypes
+
+Returns an action object used in signalling that block types by the given
+name(s) should be enabled.
```

Comment:
Can you clarify what does an enabled block type mean in this context and how it relates with other disabling mechanisms like "allowed_block_types" filter?

 - If the block is already in the content, is it rendered properly?
 - Can I transform to this kind of block?
 - Should we make exceptions for the "default block" and the fallback one? it doesn't feel right to allow disabling these blocks.

I think at the moment, the various ways of disabling blocks are confusing and we should have this documented somewhere in the handbook to allow us to have all the same understanding of these mechanisms.

Reply:
> Can you clarify what does an enabled block type mean in this context and how it relates with other disabling mechanisms like "allowed_block_types" filter?

There's more detail in the original comment. Essentially, everything here builds upon `allowed_block_types`. The preference acts as a blacklist atop the `allowed_block_types` whitelist, where the difference is calculated when rendering the `<EditorProvider>` settings from within `edit-post`. An "enabled" block type, at least in the sense of this store action, is the act of omitting a block type from this blacklist.

Reply:
The behavior of `allowed_block_types` is still unclear to me. Originally it was meant to be used to hide things from the inserter but I think it evolved into hiding the block entirely (transforms, inserter, default blocks ...) but still keep it registered for posts already using it.

Reply:
As far as I can tell, the intent is to omit is as an option everywhere where it can result in it being inserted, as you point out. They're always still registered. For posts which already contain those blocks, I'm not sure we'd care to be destructive with those "invalid" blocks anyways.